rgw/v2/tests/aws/test_tail_deletion_during_copy_object.py

```diff
@@ -0,0 +1,135 @@
+
+<input_yaml>
+    Note: Following yaml can be used
+    ceph-qe-scripts/rgw/v2/tests/aws/configs/test_tail_deletion_during_copy_object_1g.yaml
+    ceph-qe-scripts/rgw/v2/tests/aws/configs/test_tail_deletion_during_copy_object_10m.yaml
+    ceph-qe-scripts/rgw/v2/tests/aws/configs/test_tail_deletion_during_copy_object_6m.yaml
+    
+
+Operation:
+    Verifies tail object not deleted post performing copy_objject
+    GC process
+    object download is performed
+"""
+
+
+import argparse
+import json
+import logging
+import os
+import random
+import sys
+import traceback
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+
+
+from v2.lib import resource_op
+from v2.lib.aws import auth as aws_auth
+from v2.lib.aws.resource_op import AWS
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.lib.s3cmd import auth as s3_auth
+from v2.tests.aws import reusable as aws_reusable
+from v2.tests.s3_swift import reusable
+from v2.tests.s3cmd import reusable as s3cmd_reusable
+from v2.utils import utils
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+
+log = logging.getLogger(__name__)
+TEST_DATA_PATH = None
+
+
+def test_exec(config, ssh_con):
+    """
+    Executes test based on configuration passed
+    Args:
+        config(object): Test configuration
+    """
+    s3cmd_path = "/home/cephuser/venv/bin/s3cmd"
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    user_name = (config.test_ops.get("user_name"), None)
+    user_names = [user_name] if type(user_name) != list else user_name
+    user_info = resource_op.create_users(no_of_users_to_create=config.user_count)
+    for user in user_info:
+        user_name = user["user_id"]
+        ip_and_port = s3cmd_reusable.get_rgw_ip_and_port(ssh_con)
+        s3_auth.do_auth(user, ip_and_port)
+        cli_aws = AWS(ssl=config.ssl)
+        endpoint = aws_reusable.get_endpoint(ssh_con, ssl=config.ssl)
+        aws_auth.do_auth_aws(user)
+        for bc in range(config.bucket_count):
+            bucket_name = utils.gen_bucket_name_from_userid(user_name, rand_no=bc)
+            s3cmd_reusable.create_bucket(bucket_name)
+            log.info(f"Bucket {bucket_name} created")
+            log.info("Upload a file")
+            file_size = config.test_ops.get("file_size", 1024)
+            cmd = f"fallocate -l {file_size} object1"
+            utils.exec_shell_cmd(cmd)
+            cmd = f"{s3cmd_path} put object1 s3://{bucket_name}/object1"
+            utils.exec_shell_cmd(cmd)
+            utils.exec_shell_cmd(f"{s3cmd_path} ls s3://{bucket_name}")
+            out1 = aws_reusable.copy_object(cli_aws, bucket_name, "object1", endpoint)
+            utils.exec_shell_cmd("radosgw-admin gc list --include-all")
```

Comment:
I didn't understand why these GC commands are here

Reply:
Hi @TejasC88 ,
Please refer https://bugzilla.redhat.com/show_bug.cgi?id=2356678#c4,
As per BZ,
post copying source file to itself,
we can see some entries in gc list, once GC process is performed,
we were seeing issue of object not being accessible.
these step is part of reproducer:https://bugzilla.redhat.com/show_bug.cgi?id=2356678#c0

Which is currently fixed in 8.1, 8.0z5, 7.1z5

Reply:
thanks @ckulal for clarifying https://bugzilla.redhat.com/show_bug.cgi?id=2356678 has the same step